osquery/tables/networking/posix/tests/interface_ipv6_tests.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <gtest/gtest.h>
+
+#include <osquery/sql.h>
+#include <osquery/tests/test_matchers.h>
+
+namespace osquery {
+namespace tables {
+
+namespace {
+const std::set<std::string> kExpectedColumns = {
+    "interface",
+    "hlim",
+    "forwarding",
+    "redirect",
+    "rtadv",
+};
+}
+
+class InterfaceIpv6Tests : public testing::Test {};
+
+TEST_F(InterfaceIpv6Tests, test_columns_types) {
+  SQL results("select * from interface_ipv6");
+  EXPECT_TRUE(results.ok()) << results.getMessageString();
+
+  for (const auto& row : results.rows()) {
+    std::set<std::string> columns;
+    for (const auto& column : row) {
+      auto& key = column.first;
+      auto& val = column.second;
+      columns.insert(key);
+      if (key == "interface") {
+        auto intf = checkTypeAndCast<TEXT_LITERAL>(val);
+        EXPECT_THAT(intf, IsNetworkIntfName());
```

Comment:
You could also use integration tests routine to check the type and value of the results.
You could find a small example here: osquery/tests/integration/tables/user_groups.cpp
And full list of available checkers here: osquery/tests/integration/tables/helper.h

What do you think?

Reply:
This directory did not exist when I started working on the tests. The approach looks similar to what I was doing, so it'll be easy to change my code.